src/shell-integration/bash/ghostty.bash

```diff
@@ -95,6 +95,170 @@ if [[ "$GHOSTTY_SHELL_FEATURES" == *"sudo"* && -n "$TERMINFO" ]]; then
+      fi
+      builtin export "TERM_PROGRAM=ghostty"
+      ssh_opts+=(-o "SendEnv TERM_PROGRAM")
+
+      if [[ -n "$TERM_PROGRAM_VERSION" ]]; then
+        if [[ -n "${TERM_PROGRAM_VERSION+x}" ]]; then
+          ssh_exported_vars+=("TERM_PROGRAM_VERSION=${TERM_PROGRAM_VERSION}")
+        else
+          ssh_exported_vars+=("TERM_PROGRAM_VERSION")
+        fi
+        ssh_opts+=(-o "SendEnv TERM_PROGRAM_VERSION")
+      fi
+
+      ssh_env+=(
+        "COLORTERM=truecolor"
+        "TERM_PROGRAM=ghostty"
+      )
+      if [[ -n "$TERM_PROGRAM_VERSION" ]]; then
+        ssh_env+=("TERM_PROGRAM_VERSION=$TERM_PROGRAM_VERSION")
+      fi
+    fi
+
+    # Install terminfo on remote host if needed
+    if [[ "$GHOSTTY_SHELL_FEATURES" =~ ssh-terminfo ]]; then
+      builtin local ssh_config ssh_user ssh_hostname
+      ssh_config=$(builtin command ssh -G "$@" 2>/dev/null)
+
+      while IFS=' ' read -r ssh_key ssh_value; do
+        case "$ssh_key" in
+          user) ssh_user="$ssh_value" ;;
+          hostname) ssh_hostname="$ssh_value" ;;
+        esac
+        [[ -n "$ssh_user" && -n "$ssh_hostname" ]] && break
+      done <<< "$ssh_config"
+
+      ssh_target="${ssh_user}@${ssh_hostname}"
+
+      if [[ -n "$ssh_hostname" ]]; then
+        # Check if terminfo is already cached
+        builtin local ssh_cache_check_success=false
+        if builtin command -v ghostty >/dev/null 2>&1; then
+          ghostty +ssh-cache --host="$ssh_target" >/dev/null 2>&1 && ssh_cache_check_success=true
+        fi
+
+        if [[ "$ssh_cache_check_success" == "true" ]]; then
+          ssh_env+=(TERM=xterm-ghostty)
+        elif builtin command -v infocmp >/dev/null 2>&1; then
+          if ! builtin command -v base64 >/dev/null 2>&1; then
+            builtin echo "Warning: base64 command not available for terminfo installation." >&2
+            ssh_env+=(TERM=xterm-256color)
+          else
+            builtin local ssh_terminfo ssh_base64_decode_cmd
+
+            # BSD vs GNU base64 compatibility
+            if base64 --help 2>&1 | grep -q GNU; then
+              ssh_base64_decode_cmd="base64 -d"
+              ssh_terminfo=$(infocmp -0 -Q2 -q xterm-ghostty 2>/dev/null | base64 -w0 2>/dev/null)
+            else
+              ssh_base64_decode_cmd="base64 -D"
+              ssh_terminfo=$(infocmp -0 -Q2 -q xterm-ghostty 2>/dev/null | base64 2>/dev/null | tr -d '\n')
+            fi
+
+            if [[ -n "$ssh_terminfo" ]]; then
+              builtin echo "Setting up Ghostty terminfo on $ssh_hostname..." >&2
+              builtin local ssh_cpath_dir ssh_cpath
+
+              ssh_cpath_dir=$(mktemp -d "/tmp/ghostty-ssh-$ssh_user.XXXXXX" 2>/dev/null) || ssh_cpath_dir="/tmp/ghostty-ssh-$ssh_user.$$"
+              ssh_cpath="$ssh_cpath_dir/socket"
+
+              if builtin echo "$ssh_terminfo" | $ssh_base64_decode_cmd | builtin command ssh "${ssh_opts[@]}" -o ControlMaster=yes -o ControlPath="$ssh_cpath" -o ControlPersist=60s "$@" '
```

Comment:
Here we decode the base64-encoded terminfo string that we just formed above. Do we need to base64-encoded it at all if it's immediately sent in its unencoded form?

Reply:
Ah, the `-Q` option was introduced in a later ncurses version (installable from homebrew on macOS):

```
-Q number  dump compiled description
```

On ncurses 6.5, that `-Q2` actually prints its output in base64, so I think there's still some mismatch between what different systems can produce (based on what I'm seeing and what you're seeing).

I think that reenforces the argument that we should stick with just `-0 -x` for compatibility.